### ISSUE TITLE
Ability to provide custom icons for applications

### ISSUE THREAD

**Issue by Nuru** [gravitational/teleport#4712]
### Feature Request
The Teleport Application Access Proxy (AAP) provides a "dashboard" listing applications that can be accessed via the proxy. Currently (5.0.0.beta.9) every application is represented by the same 62px square icon (actually font glyph). Users should be able to specify an application-specific icon to use (via URL) for each application. 

### Motivation
If all the applications have the same icon, then the icon is useless in helping guide the user to find the right application. This should be a trivial to provide an image element with a user-specified URL.

### Who's it for?
OSS User, Pro, Enterprise

**Comment by shamil** [comment]
We are also interested in this ability.

**Comment by MateoSosa01** [comment]
Would be great if the webapps favicon is passed through as the icon in teleport GUI as an automatic process.
In addition to that it'd be great to provide custom icons.

**Comment by adberger** [comment]
Will this be possible in the near future or should we do PRs for additional application icons?  
We currently have the need for:
- Prometheus
- Hubble (Cilium)
- WeaveGitOps
- Kubernetes Dashboard

**Comment by zagr0** [comment]
The strange thing is that for grafana icon is displayed, not sure if that's something hard coded, but would be great to have favicons to be showed automatically as application icons.

**Comment by adberger** [comment]
> The strange thing is that for grafana icon is displayed, not sure if that's something hard coded, but would be great to have favicons to be showed automatically as application icons.

Yes, Grafana among other applications are hardcoded to be displayed correctly.
Probably in here: https://github.com/gravitational/teleport/blob/89d1aa460d9c4065feeb522e909791f87cb41104/web/packages/shared/components/UnifiedResources/shared/viewItemsFactory.ts#L178

**Comment by elijahnyp** [comment]
This would be a fantastic feature - ideally just have the app_service scrape the favicon and use that - with a key in the yaml resource and annotation to enable/disable the feature.  It would make the GUI significantly easier to navigate and would make it look a LOT more friendly.

**Comment by shahar-cortanix** [comment]
since this issue was opened in 2020, will we get some temp solution? maybe a quick fix / hack for just manually selecting or uploading? without the auto-guess application and automatic iconing (which is cool but will take forever)

**Comment by cmeury** [comment]
We are new Teleport customers, and we would love this feature. It would dramatically improve the UX for anyone using the visual interface regularly.

**Comment by malahayatii** [comment]
Since there wasn't much input on this front for a little while, I would like to bring up a question from earlier this year. Should we create a PR for additional application icons we would like to see incorporated? Or is something already in the works.

**Comment by pschisa** [comment]
+1 for bringing in customizable web app icons

**Comment by uedvt359** [comment]
There are many false positives with the current approach of guessing the icon from the hostname. For example, both my netbox and sandbox servers get the `box.svg` icon, despite neither being related with box.com in any way. or any server/application hostname that just happens to contain `g2` (e.g. `thing2.example.com` - even worse, since `thing1` now has a different icon, despite being the same service)

**Comment by ev1lm0nk3y** [comment]
I liked this idea and have a sample PR within my fork that will allow for apps to define the icon they want in the webui. https://github.com/ev1lm0nk3y/teleport/pull/1 

This is a suggestion, but I think it would resolve this request

**Comment by praveenprem** [comment]
+1 for bringing in customizable web app icons. 
> Customer's ideal scenario is having the ability to provide custom icon they can choose but happy if there are standard well knows icons are available by default as well.

**Comment by davmar59** [comment]
+1 as well for customizable web app icons
